<?xml version="1.0" encoding="UTF-8"?> 
<launch> 
    <param name="/use_sim_time" value="true"/>

        <!-- <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera/color/image_raw raw out:=/camera/image_raw" /> -->
    
    <arg name="vocabulary" default="/root/catkin_ws/src/ORB_SLAM3/Vocabulary/ORBvoc.txt" doc="Vocabulary for DBoW2."/> 
    <arg name="settings" default="/root/catkin_ws/src/ORB_SLAM3/Examples/ROS/ORB_SLAM3/config/agro_sim_mono_inertial.yaml" doc="System settings."/> 
    <arg name="map_frame_id" default="world"/>
    <arg name="base_link_frame_id" default="base_link"/>
    <node name="orbslam3_mono_inertial" pkg="ORB_SLAM3" type="Mono_Inertial" output="screen" args="$(arg vocabulary) $(arg settings) true"> 
        <remap from="/camera/image_raw" to="/camera/color/image_raw"/>
        <remap from="/imu" to="/camera/imu/sample"/>
        <param name="map_frame_id" value="$(arg map_frame_id)"/>
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)"/>
    </node>

     <node pkg="tf2_ros" type="static_transform_publisher" name="start_broadcaster" args="0 0 0 0 0 0 start map"/>
</launch>
